### ISSUE TITLE
Can't get callback to work

### ISSUE THREAD

**Issue by some-mthfka** [adolenc/cl-neovim#17]
Hi! I can't get to `call/s` the function that I defined without getting an error:
```
(nvim:enable-logging :level :debug)
```
```
CL-USER> (nvim:defun/s "my_test_function" (x y) (print (list x y)))
 <DEBUG> [22:42:25] cl-neovim - (cl-neovim -> nvim) [req] (80 "vim_call_function" ("lisp#RegisterRepl" (3)))
 <DEBUG> [22:42:25] cl-neovim - (nvim -> cl-neovim) [rsp] (1 80 NIL 0)
 <DEBUG> [22:42:25] cl-neovim - (cl-neovim -> nvim) [req] (81 "vim_call_function" ("lisp#RegisterReplCallback" (3 #<HASH-TABLE :TEST EQUAL :COUNT 4 {1005333B33}>)))
 <DEBUG> [22:42:25] cl-neovim - (nvim -> cl-neovim) [rsp] (1 81 NIL 0)
 <DEBUG> [22:42:25] cl-neovim - Registering callback `":function:my_test_function"'
#<FUNCTION (LAMBDA (&REST #:R805)) {52FD8F4B}>
CL-USER> (nvim:call/s nvim:*nvim-instance* "my_test_function" 1 2)
 <DEBUG> [22:42:37] cl-neovim - (cl-neovim -> nvim) [req] (82 "my_test_function" (1 2))
 <DEBUG> [22:42:37] cl-neovim - (nvim -> cl-neovim) [rsp] (1 82 (0 "Invalid method: my_test_function") NIL)
; Debugger entered on #<MESSAGEPACK-RPC.CONDITIONS:RPC-ERROR {100533EE93}>
```
The setup, though, seems to work otherwise, I can call the already defined functions just fine and
```
:Lisp (print 42)
```
prints 42 in nvim as expected.

Any clue what this could be?

**Comment by adolenc** [comment]
Sup mthfka,

I see two potential issues:

1. When defining function with the string name, it has to adhere to Neovim's naming rules. Specifically (from `:h function`):
> The name must be made of alphanumeric characters and '_', and must start with a capital
2. If I remember correctly, `nvim:call` is meant more for low-level interactions with Neovim api, whereas here you want to call a function you defined in Neovim. For that, you should probably use `nvim:funcall` instead (which [is just a nice abstraction over nvim:call/s for calling functions](https://github.com/adolenc/cl-neovim/blob/7212d305206aaae331a3e2d0d2597b671cec01f4/src/manual-api.lisp#L30)).

In short, the following works for me:
```
(nvim:defun/s "MyTestFunction" (x y) (print (list x y)))
(nvim:funcall "MyTestFunction" 1 2)
```

**Comment by some-mthfka** [comment]
Huh, yeah, that works!

Initially I was trying to subscribe to one of the UI callbacks, like below, but it appears I still can't get it to work, even with the correct name, nothing is happening. This looks like a strictly neovim-side issue to solve from now on, meaning I am not subscribing correctly, but, just as a shot in the dark, could you please glance over this output as well?

```
CL-USER> (nvim:defun/s "MyTestFunction" (x y) (print (list x y)))
#<FUNCTION (LAMBDA (&REST #:R672)) {52FC618B}>
CL-USER> (nvim:funcall "MyTestFunction" 1 2)

(1 2) 
(1 2)
CL-USER> (nvim:enable-logging :level :debug)
:DEBUG
CL-USER> (nvim:subscribe "cursor_goto" "MyTestFunction")
 <DEBUG> [00:10:01] cl-neovim - Registering callback `"cursor_goto"'
 <DEBUG> [00:10:01] cl-neovim - (cl-neovim -> nvim) [req] (5 "vim_subscribe" ("cursor_goto"))
 <DEBUG> [00:10:01] cl-neovim - (nvim -> cl-neovim) [rsp] (1 5 NIL NIL)
NIL
CL-USER> (nvim:ui-attach 20 20 #())
```
```
CL-USER> (nvim:input "i some text")
 <DEBUG> [00:17:19] cl-neovim - (cl-neovim -> nvim) [req] (15 "nvim_input" ("i some text"))
 <DEBUG> [00:17:19] cl-neovim - (nvim -> cl-neovim) [ntf] (2 "redraw" (("cursor_goto" (19 9)) ("highlight_set" (#<HASH-TABLE :TEST EQUALP :COUNT 0 {1004ED96B3}>)) ("put" ("0")) ("cursor_goto" (0 5)) ("flush" NIL)))
 <DEBUG> [00:17:19] cl-neovim - (nvim -> cl-neovim) [ntf] (2 "redraw" (("cursor_goto" (19 9)) ("put" (" ")) ("cursor_goto" (18 12)) ("highlight_set" (#<HASH-TABLE :TEST EQUALP :COUNT 2 {1004EDAEB3}>)) ("put" ("1")) ("cursor_goto" (0 0)) ("flush" NIL)))
 <DEBUG> [00:17:19] cl-neovim - (nvim -> cl-neovim) [rsp] (1 15 NIL 11)
```

**Comment by some-mthfka** [comment]
Oh, wait, if I subscribe a lisp function to `redraw`, the callback works. Thanks for the fast feedback on the funcall fix!
